src/java.base/share/classes/sun/security/x509/PolicyInformation.java

```diff
@@ -122,28 +122,22 @@ public PolicyInformation(DerValue val) throws IOException {
     /**
      * Compare this PolicyInformation with another object for equality
      *
-     * @param other object to be compared with this
+     * @param obj object to be compared with this
      * @return true iff the PolicyInformation objects match
      */
-    public boolean equals(Object other) {
-        if (!(other instanceof PolicyInformation piOther))
-            return false;
-
-        if (!policyIdentifier.equals(piOther.getPolicyIdentifier()))
-            return false;
-
-        return policyQualifiers.equals(piOther.getPolicyQualifiers());
+    @Override
+    public boolean equals(Object obj) {
+        return obj instanceof PolicyInformation other
+                && policyIdentifier.equals(other.getPolicyIdentifier())
+                && policyQualifiers.equals(other.getPolicyQualifiers());
     }
 
     /**
-     * Returns the hash code for this PolicyInformation.
-     *
-     * @return a hash code value.
+     * {@return the hash code for this PolicyInformation}
      */
+    @Override
     public int hashCode() {
-        int myhash = 37 + policyIdentifier.hashCode();
-        myhash = 37 * myhash + policyQualifiers.hashCode();
-        return myhash;
+        return Objects.hash(policyIdentifier, policyQualifiers);
```

Comment:
What is the criteria for changing the hashCode() impl? It seems that you change some and not others. With the many hashCode() impls (e.g. Objects.hash(), Arrays.hashCode()), I wonder why this and not that.

Reply:
Whether I suggest changing hashCode or not depends on multiple criteria, of which these two are the most important:

  * Is the value specified or relied upon?
  * Does it need to be performant?

If the answer to both of these is "NO", I might refactor hashCode. One unintended side effect, is that sometimes such a refactoring may improve hashCode quality (think: perfect hashing) almost for free.

If you have concerns on particular refactorings, I'd be happy to discuss them.

Reply:
I would add to that list that I think it is really important any change to the `hashCode` impl doesn't break the equals/hashCode contract, so it's important to look at the `equals` implementation in conjunction with any changes to `hashCode`, mostly to ensure 2 equal objects continue to have the same hash code. It is also important to question any oddity about the `hashCode` impl, like I did for DESKey and try to understand why it may have been implemented that way, such as skipping the first byte.

Reply:
True. I skipped that criteria, as I thought it should go without saying.

Reply:
> I would add to that list that I think it is really important any change to the `hashCode` impl doesn't break the equals/hashCode contract, so it's important to look at the `equals` implementation in conjunction with any changes to `hashCode`, mostly to ensure 2 equal objects continue to have the same hash code.

True. I skipped that criteria, as I thought it should go without saying.

> It is also important to question any oddity about the `hashCode` impl, like I did for DESKey and try to understand why it may have been implemented that way, such as skipping the first byte.

True again. If you have any other examples where the changes to hashCode look suspicious, please tell me so.